src/app.jsx

```diff
@@ -357,3 +363,124 @@ export const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }) => {
         </Select>
     );
 };
+
+const DropdownWithKebab = ({ selected, path }) => {
+    const Dialogs = useDialogs();
+    const [isOpen, setIsOpen] = useState(false);
+
+    const onToggleClick = () => {
+        setIsOpen(!isOpen);
+    };
+    const onSelect = (_event, itemId) => {
+        setIsOpen(false);
+    };
+
+    const deleteItem = () => {
+        const itemPath = "/" + path.join("/") + "/" + selected.name;
+        Dialogs.show(<ConfirmDeletionDialog selected={selected} itemPath={itemPath} />);
+    };
+
+    return (
+        <Dropdown
+            isPlain
+            isOpen={isOpen}
+            onSelect={onSelect}
+            onOpenChange={isOpen => setIsOpen(isOpen)}
+            popperProps={{ position: "right" }}
+            toggle={toggleRef =>
+                <MenuToggle ref={toggleRef} variant="plain" onClick={onToggleClick} isExpanded={isOpen}>
+                    <EllipsisVIcon />
+                </MenuToggle>}
+        >
+            <DropdownList>
+                <DropdownItem itemId='delete-item' key="delete-item" isDisabled={!selected} onClick={deleteItem} className={!selected ? "" : "pf-m-danger"}>
+                    {_("Delete")}
+                </DropdownItem>
+            </DropdownList>
+        </Dropdown>
+    );
+};
+
+const ConfirmDeletionDialog = ({ selected, itemPath }) => {
+    const Dialogs = useDialogs();
+
+    const deleteItem = () => {
+        if (selected.type === "file") {
+            cockpit.spawn(["rm", itemPath]).then(Dialogs.close)
+                    .catch(() => { Dialogs.show(<ForceDeleteModal itemPath={itemPath} />) });
+        } else if (selected.type === "directory") {
+            let emptyDirectory = false;
+            // Check for empty directory
+            cockpit.spawn(["ls", "-A", itemPath])
+                    .then(result => { emptyDirectory = result === "" })
+                    .then(() => {
+                        if (emptyDirectory)
+                            cockpit.spawn(["rmdir", itemPath]).then(Dialogs.close);
+                        else {
+                            cockpit.spawn(["rm", "-r", itemPath]);
+                        }
+                    })
+                    .then(Dialogs.close)
+                    .catch(() => { Dialogs.show(<ForceDeleteModal itemPath={itemPath} />) });
+        }
+    };
+
+    return (
+        <Modal
+            position="top"
+            title={_("Confirm deletion")}
```

Comment:
Use warning modal variant like here. https://patternfly-react-v5.surge.sh/components/modal/#title-icon

And also maybe adjust the title to contain the directory/file name?

`cockpit.format(_("Delete $0 $1?", selected.name, selected.type == "file" ? _("file") : _("directory") ))`

Reply:
I switched this to `_(cockpit.format("Delete $0 $1?", selected.name, selected.type === "file" ? _("file") : _("directory")))` (put the translation function on the outside), since the code you mentioned didn't display the correct string

Reply:
This is defnitely not correct. just conditionalize and have two variants.
if (selected.type == 'file') return cockpit.format(_("Delete $0 file?", selected.name); else...